experimental/PlaneCurve/EllCurveZnZ.jl

```diff
@@ -0,0 +1,704 @@
+export sum_Point_EllCurveZnZ,
+    ECM,
+    rand_pair_EllCurve_Point,
+    IntMult_Point_EllCurveZnZ,
+    cornacchia_algorithm,
+    Miller_Rabin_test,
+    Pollard_rho,
+    Pollard_p_1,
+    ECPP
+
+################################################################################
+# Elliptic curves over a ring Z/nZ
+################################################################################
+################################################################################
+# Helping functions
+################################################################################
+# addition on two projective points on an elliptic curve over Z/nZ, given in
+# short Weierstrass form, when it is possible. It gives as an output a pair
+# composed of the coordinates of the sum and ZZ(1) if the sum exists, and
+# (0, 0, 0) and the gcd otherwise.
+# Adapted from the corresponding code in Hecke.
+
+function _add(
+    P::Array{Nemo.fmpz_mod,1},
+    Q::Array{Nemo.fmpz_mod,1},
+    E::Array{Nemo.fmpz_mod,1},
+)
+    length(P) == 3 && length(Q) == 3 || error("arrays of size 3 required")
+    length(E) == 2 || error("array of size 2 required")
```

Comment:
So this is *not* a change request for this PR (it would be too big a change, I think), just a general remark: whenever you find yourself dealign with vectors which have small constant length (such as 2 and 3 here), it might be more appropriate to use tuples instead. So  here you could use `NTuple{3, Nemo.fmpz_mod}` which is shorthand for `Tuple{Nemo.fmpz_mod, Nemo.fmpz_mod, Nemo.fmpz_mod}`:

```suggestion
function _add(
    P::NTuple{3, Nemo.fmpz_mod}
    Q::NTuple{3, Nemo.fmpz_mod}
    E::NTuple{2, Nemo.fmpz_mod}
)
```